src/ufo-read-task.c

```diff
@@ -258,69 +261,73 @@ ufo_read_task_get_requisition (UfoTask *task,
 {
     UfoReadTaskPrivate *priv;
     const gchar *filename;
+    gsize num_images = 0;
 
     priv = UFO_READ_TASK_GET_PRIVATE (UFO_READ_TASK (task));
 
-    if (priv->reader == NULL) {
-        filename = (gchar *) priv->current_element->data;
-        priv->reader = get_reader (priv, filename);
-
-        if (!ufo_reader_open (priv->reader, filename, priv->start, error))
-            return;
-
-        priv->start = 0;
-    }
-
-    if (!ufo_reader_data_available (priv->reader)) {
+    if (!priv->reader || !ufo_reader_data_available (priv->reader)) {
         GList *last_element;
         guint tries;
 
-        ufo_reader_close (priv->reader);
-        last_element = priv->current_element;
-        priv->current_element = g_list_nth (priv->current_element, priv->step);
-
-        if (priv->current_element == NULL) {
-            if (priv->retries == 0 || priv->current == priv->number) {
-                priv->done = TRUE;
-                priv->reader = NULL;
-                return;
+        while (TRUE) {
+            /* Keep skipping files until we find one with enough images to start
+             * reading at priv->image_start index. */
+            if (priv->reader) {
+                ufo_reader_close (priv->reader);
+                last_element = priv->current_element;
+                priv->current_element = g_list_nth (priv->current_element, priv->step);
             }
 
-            for (tries = 0; tries < priv->retries && priv->current_element == NULL; tries++) {
-                 GList *new_list;
-                 GList *match;
-
-                 g_debug ("read: retry %i/%i, waiting %is for new files", tries + 1, priv->retries, priv->retry_timeout);
-                 g_usleep (priv->retry_timeout * G_USEC_PER_SEC);
-                 new_list = g_list_sort (read_filenames (priv), (GCompareFunc) g_strcmp0);
-                 match = g_list_find_custom (new_list, last_element->data, (GCompareFunc) g_strcmp0);
-
-                 if (match != g_list_last (new_list)) {
-                     g_list_free_full (priv->filenames, (GDestroyNotify) g_free);
-                     priv->filenames = new_list;
-                     priv->current_element = g_list_next (match);
-                 }
-                 else {
-                     g_list_free_full (new_list, (GDestroyNotify) g_free);
-                 }
+            if (priv->current_element == NULL) {
+                if (priv->retries == 0 || priv->current == priv->number) {
+                    priv->done = TRUE;
+                    priv->reader = NULL;
+                    return;
+                }
+
+                for (tries = 0; tries < priv->retries && priv->current_element == NULL; tries++) {
```

Comment:
Is anyone actually using that at all? I remember implementing that for a particular use case but don't remember anyone using that. It always felt a bit icky to wait for the appearance of new files instead of using any of the notification APIs.

Reply:
That's exactly what I thought. I think we will actually never need this. Delete? And why didn't you use the notification APIs?

Reply:
> Delete? And why didn't you use the notification APIs?

Personally, I'd just remove it yes. And I did not use them probably because it was just a quick and dirty solution for a quick and dirty experiment ;-)

Reply:
Et voila a75fe709b9d3066af22c3d54c8fb1066afd7e4cc